iroh-net/src/magicsock/endpoint/best_addr.rs

```diff
@@ -0,0 +1,202 @@
+//! The [`BestAddr`] is the currently active best address for UDP sends.
+
+use std::{
+    net::SocketAddr,
+    time::{Duration, Instant},
+};
+
+use iroh_metrics::inc;
+use tracing::{debug, info, trace};
+
+use super::{AddrLatency, TRUST_UDP_ADDR_DURATION};
+use crate::magicsock::metrics::Metrics as MagicsockMetrics;
+
+#[derive(Debug, Default)]
+pub(super) struct BestAddr(Option<BestAddrInner>);
+
+#[derive(Debug)]
+struct BestAddrInner {
+    addr: AddrLatency,
+    trust_until: Option<Instant>,
+    confirmed_at: Instant,
+}
+
+#[derive(Debug)]
+pub(super) enum BestAddrSource {
+    ReceivedPong,
+    BestCandidate,
+}
+
+impl BestAddrSource {
+    fn trust_until(&self, from: Instant) -> Instant {
+        match self {
+            BestAddrSource::ReceivedPong => from + TRUST_UDP_ADDR_DURATION,
+            // TODO: Fix time
+            BestAddrSource::BestCandidate => from + Duration::from_secs(60 * 60),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub(super) enum BestAddrState<'a> {
+    Valid(&'a AddrLatency),
+    Outdated(&'a AddrLatency),
+    Empty,
+}
+
+#[derive(Debug)]
+pub enum BestAddrClearReason {
```

Comment:
just `ClearReason`,  to avoid `best_addr::BestAddrClearReson` double

Reply:
fixed!